internal/gensample/plugin.go

```diff
@@ -27,7 +27,11 @@ const (
 	paramError = "need parameter in format: go-gapic-package=client/import/path;packageName"
 )
 
-// PluginEntry is the entry point of SampleGen as a protoc plugin, together with CodeGen
+// PluginEntry is the entry point of SampleGen as a protoc plugin.
+// When running gapic-generator-go as a docker image or a protoc plugin, both gapic clients
```

Comment:
The explanation is very helpful. Thanks!

However, I am uncomfortable including details here of this function's callers. It seems like this raises the probability this comment will go stale.

How about saying here that if a protoc plugin is invoked that wishes to generate samples, it will eventually call this function to do so?

And then move the actual specifics you currently have (the configuration presence as a signal, the `main()` filename) to the main plugin source file and/or a README or other doc that summarizes the code layout.

Reply:
Done. I'll update readme separately.